ldms/src/ldmsd/ldmsd_request.c

```diff
@@ -5679,8 +5681,8 @@ static char *__xprt_stats_as_json(size_t *json_sz)
 					    (struct sockaddr *)&ss_local,
 					    (struct sockaddr *)&ss_remote,
 					    &socklen);
-			strncpy(xprt_type, ldms_xprt_type_name(op->op_min_xprt),
```

Comment:
What 's the warning that you got here?

Reply:
It warned about the combination of size and string not terminating with '\0'.

Reply:
I don't quite get that. The man page says that the result will always be terminated with a '\0' What are we missing here.

Reply:
From `strncpy(3)`

>The  strncpy()  function is similar, except that at most n bytes of src
>are copied.  Warning: If there is no null byte among the first n  bytes
>of src, the string placed in dest will **not** be null-terminated.

Reply:
I feel like we're going round in circles here. This used to be strcpy, and then someone changed a bunch of these to strncpy because some tool was warning about strcpy.  So now we're going to snprintf because yet another tool is warning about the potentially missing '\0'.  strncpy is less efficient, writing "" to a 1MB buffer will write 1MB of '\0' to the destination, and snprintf is slower still. These changes are all non-functional, progressively less efficient, and succeed only in quieting a warning about a non-issue. What platform is this on? I'm leaning towards just quieting that warning globally on that platform and calling it a day.

Reply:
I agree with you. Shall we use `memcpy()` and suppress the warning if it came up instead? We already know the lengths of both source and destination.

Reply:
If we know the length of the source and the destination, then why isn't strcpy sufficient? Does gcc complain about strcpy? Also, I see a bunch of places where someone is using strncpy, but the length is dependent on the *source* buffer, not the destination. Those changes were no doubt made due to a warning and created a bug in the process. Does every use of strncpy create this warning, or only some. And if it's only some, why?